### PR TITLE
LEARNER-1937 Add ability to extract and compile translation strings

### PR DESCRIPTION
**Pull request by amangano-edx**
Adds make targets and requirements necessary to extract and compile translation strings.

@edx/learner-heart 

https://openedx.atlassian.net/browse/LEARNER-1937

### CODE REVIEW

**Comment by clintonb** [comment]
Add documentation explaining how to use these commands, and what to do with the extracted files.

**Comment by amangano-edx** [comment]
@clintonb I just updated the Readme, please take another look when you have a second.

File: README.rst [hunk 1]
```diff
@@ -12,3 +12,19 @@ Building
 
 Build the assets by running `make build`. Compiled assets should be *committed* to the repository so that they are
 accessible by the Credentials Service when the package is installed.
+
+i18n
+-----
+
+Run `make i18n_requirements` to install dependencies necessary for running i18n commands. Note: this will install
+django and is best done from within a python virtualenv.
```
Comment (clintonb) [inline README.rst]: s/django/Django
s/python/Python

File: README.rst [hunk 2]
```diff
@@ -12,3 +12,19 @@ Building
...
+
+To mark strings in templates/partials for translation, simply wrap them in one of django's built-in translation functions.
```
Comment (clintonb) [inline README.rst]: s/django/Django

File: conf/locale/requirements.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+django==1.11.2
```
Comment (clintonb) [inline conf/locale/requirements.txt]: `django>=1.11.2,<2.0`

**Comment by amangano-edx** [comment]
@clintonb travis keeps failing due to a difference in how package.json gets created during the build. I don't think that was the intention of the `make test` command, so I've updated it to only diff the contents of the edx_credentials_themes directory. Does that seem reasonable to you?

**Comment by clintonb** [comment]
That's fine. Alternatively, just run `npm install` with Node 8/npm 5 to update the file.

**Pull request merged**